### ISSUE TITLE
ZnapZend throws internal error when attempting to send snapshots since 20.09

### ISSUE THREAD

**Issue by kaldonir** [NixOS/nixpkgs#106662]
**Describe the bug**
Since 20.09, ZnapZend fails to transfer snapshots for subsequent datasets if it encounters an error with one. This worked in 20.03, but now it throws an internal ZnapZend error. A workaround is to set `logTo` to a file, e.g., `"/var/log/znapzend"`, but this causes logs to no longer be accessible via `journalctl`.

**To Reproduce**

I am not sure how exactly this can be reproduced, but here is how everything is set up:

- `zfs list`:
```
NAME                USED  AVAIL     REFER  MOUNTPOINT
datapool            541G  2.98T       24K  /mnt/datapool
datapool/artifact   122G  2.98T      121G  legacy
datapool/docker     418G  2.98T     3.95G  legacy
```

- NixOS config for `services.znapzend`:
```nix
{
  enable = true;
  zetup = {
    "datapool" = {
      # local plan (required)
      plan = "25h=>1h";
      recursive = true;
      destinations.remote = {
        host = "server1_zfs_user@server2";
        dataset = "backup/server1-znapzend";
        plan = "1d=>1h,1w=>1d,1m=>1w,1y=>1m,10y=>1y";
      };
    };
  };
}
```
- Failing log output:
```
Dec 03 12:00:03 server1 znapzend[5069]: send/receive worker for datapool done (7339)
Dec 03 13:00:00 server1 znapzend[5069]: snapshot worker for datapool spawned (7442)
Dec 03 13:00:00 server1 znapzend[7442]: creating recursive snapshot on datapool
Dec 03 13:00:00 server1 znapzend[7442]: checking for explicitly excluded ZFS dependent datasets under 'datapool'
Dec 03 13:00:00 server1 znapzend[5069]: snapshot worker for datapool done (7442)
Dec 03 13:00:00 server1 znapzend[5069]: send/receive worker for datapool spawned (7537)
Dec 03 13:00:00 server1 znapzend[7537]: starting work on backupSet datapool
Dec 03 13:00:00 server1 znapzend[7537]: sending snapshots from datapool to server1_zfs_user@server2:backup/server1-znapzend
Dec 03 13:00:03 server1 znapzend[7549]: umount: only root can use "--types" option
Dec 03 13:00:03 server1 znapzend[7549]: cannot unmount '/backup/server1-znapzend/docker': umount failed
Dec 03 13:00:03 server1 znapzend[7537]: ERROR: cannot send snapshots to backup/server1-znapzend on server1_zfs_user@server2
Dec 03 13:00:03 server1 znapzend[5069]: send/receive for datapool failed: Can't call method "message" without a package or object reference at /nix/store/bzaav9wf216d431w0nf3jpvxcz3kg83g-znapzend-0.20.0/lib/ZnapZend.pm line 369.
Dec 03 13:00:03 server1 znapzend[5069]: send/receive worker for datapool done (7537)
```
  Where the second to last line is the internal ZnapZend error.

- The `ERROR: cannot send snapshots` message occured on 20.03 as well, but it continued sending the child datasets:
```
Dec 03 08:00:02 server1 znapzend[26426]: ERROR: cannot send snapshots to backup/server1-znapzend on server1_zfs_user@server2
Dec 03 08:00:02 server1 znapzend[26426]: sending snapshots from datapool/artifact to server1_zfs_user@server2:backup/server1-znapzend/artifact
Dec 03 08:00:05 server1 znapzend[26426]: sending snapshots from datapool/docker to server1_zfs_user@server2:backup/server1-znapzend/docker
Dec 03 08:00:07 server1 znapzend[26426]: ERROR: suspending cleanup source dataset because at least one send task failed
Dec 03 08:00:07 server1 znapzend[26426]: done with backupset datapool in 7 seconds
Dec 03 08:00:07 server1 znapzend[34998]: send/receive worker for datapool done (26426)
```
- This might be triggered by this part of the error message:
```
Dec 03 13:00:03 server1 znapzend[7549]: umount: only root can use "--types" option
Dec 03 13:00:03 server1 znapzend[7549]: cannot unmount '/backup/server1-znapzend/docker': umount failed
```
But in any case, I'd expect znapzend to behave like it did in 20.03.

**Expected behavior**
Znapzend should not fail to send snapshots because of an internal error if configured with NixOS default settings. Log messages should be accessible through `journaltcl`.

**Notify maintainers**
@otwieracz @infinisil @SlothOfAnarchy

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.80, NixOS, 20.09pre-git (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-20.03.2107.70717a337f7"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - znapzend
# a list of nixos modules affected by the problem
module:
  - znapzend
```

**Comment by Infinisil** [comment]
This sounds like an upstream issue. Try using e.g. `znapzend --runonce=datapool` to reproduce the problem without involvement of the NixOS module. If that also causes the issue (which I strongly suspect it should), then this issue should probably be reported (and searched for) in https://github.com/oetiker/znapzend/issues instead

**Comment by kaldonir** [comment]
> This sounds like an upstream issue. Try using e.g. `znapzend --runonce=datapool` to reproduce the problem without involvement of the NixOS module.

It shows similar behavior, but does not print an error message to stdout when called with `--logto=syscall::daemon`. I have opened an issue at https://github.com/oetiker/znapzend/issues/534

**Comment by blitz** [comment]
@kaldonir The upstream bug was fixed in  v0.21.0. Can we close this issue once znapzend in nixpkgs is bumped?
